### PR TITLE
[AMQPConnection] Add an ability to publish a message

### PR DESCRIPTION
**Pull request by ashie**
AMQPConnection (which is a wrapper class for rabbitmq-c) doesn't have an ability to send a message.
@cosmo0920 and I added an ad-hoc implementation of it.
Although I'll heavily modify it later to introduce an abstraction layer of the communication,
I want to merge the current code to implement the skeleton of the server side of HAPI-2.0 ASAP.

### CODE REVIEW

**Comment by cosmo0920** [comment]
:green_heart:

File: server/src/AMQPConnectionInfo.cc [hunk 1]
```diff
@@ -32,6 +34,7 @@ using namespace mlpl;
 struct AMQPConnectionInfo::Impl {
 	Impl()
 	: m_URL(),
+	  m_URLBuf(NULL),
```
Comment (kz0817) [inline server/src/AMQPConnectionInfo.cc]: m_URLBuf is only used in setURL(), isn't it.
If so, it's unnecessary to be an instance variable.
Or will you use it in the methods you are going to make later.

Comment (ashie) [inline server/src/AMQPConnectionInfo.cc]: > m_URLBuf is only used in setURL(), isn't it.
> If so, it's unnecessary to be an instance variable.

No, it's used by m_parsedURL.

It's a labyrinth of rabbitmq-c.
This buffer is referred from m_parsedURL (amqp_connection_info).
So we need to keep it until m_parsedURL is destroyed.

Comment (ashie) [inline server/src/AMQPConnectionInfo.cc]: amqp_parse_url(url, connectionInfo) do the following things:

* Replace some characters in url with "\0" to split the url string to url elements
* Store pointer to each url elements into connectionInfo. It doesn't copy the strings, refer them directly.

Comment (kz0817) [inline server/src/AMQPConnectionInfo.cc]: OK. I see.
But we want to recognize the relationship easily.
Anyway,  I'll merge this PR.

Comment (ashie) [inline server/src/AMQPConnectionInfo.cc]: > But we want to recognize the relationship easily.

I agree with you.
I'll add a comment or something in another PR.

**Pull request merged**